packages/core/test/utils_test.js

```diff
@@ -3843,6 +3843,33 @@ describe("utils", () => {
         }))
       );
     });
+    it("should infer correct anyOf schema based on data if passing undefined", () => {
```

Comment:
Let's add a test for getMatchingOption
- where formData is `null`
- Where option 2 is `{type: null}`
- We expect `getMatchingOption` to be equal to 2



```
options = [{type: "string"}, {type: "string"}, {type: "null"}];
expect(getMatchingOption(null, options, rootSchema)).eql(2);
```